cmd/snap/cmd_run.go

```diff
@@ -100,6 +105,9 @@ func (x *cmdRun) Execute(args []string) error {
 		// TRANSLATORS: %q is the hook name; %s a space-separated list of extra arguments
 		return fmt.Errorf(i18n.G("too many arguments for hook %q: %s"), x.HookName, strings.Join(args, " "))
 	}
+	if x.Timer != "" && (x.HookName != "" || x.Command != "") {
+		return fmt.Errorf(i18n.G("cannot use --timer with either --hook or --command"))
```

Comment:
hmm... either with negatives is weird, but I can't find a non-weird way of saying this. And you've already checked that hookname and command aren't both set, above. So, either:
* make a single check along the lines of `"you can only use one of --hook, --command, and --timer"`, or
* check which one it is here, and have two errors, one for each case.

I think the first one would be a little bit better (as it covers the case where the user used the three)